cmd/monaco/generate/deletefile/deletefile.go

```diff
@@ -185,6 +189,28 @@ func skipping(ttype string, included, excluded map[string]struct{}) bool {
 	return false
 }
 
+func toMapKey(d persistence.DeleteEntry) string {
+	values := make(map[string]string)
+	values["project"] = d.Project
+	values["type"] = d.Type
+	values["configId"] = d.ConfigId
+	values["configName"] = d.ConfigName
+	values["scope"] = d.Scope
+	maps.Copy(values, d.CustomValues)
+	j, _ := json.Marshal(sortedKeyValues(values)) //nolint:errchkjson
```

Comment:
I know @arthurpitman suggested this, but: WHY?

We're now building a map, copying over the entries from CustomValues, sorting it and finally marshalling to JSON.
So just to get a unique string key to use for a map, we add the overhead of allocating a map and marshaling to JSON - where before we simply rendered the string with fmt.Sprintf.

I'm missing the benefit this refactoring provides tbh

Reply:
more specifically: 
```
func TestClash(t *testing.T) {
	type My struct {A, B string}
	my := My{A: "h/", B: "i"}
	my2 := My{A: "h", B: "/i"}
	fmt.Printf("%s/%s\n", my.A, my.B)   // Output: h//i
	fmt.Printf("%s/%s\n", my2.A, my2.B) // Output: h//i
}

```

Reply:
I'm missing how 
```
fmt.Sprintf("%s/%s/%s/%s/%s/%s", d.Project, d.Type, d.ConfigId, d.ConfigName, d.Scope, sortedKeyValues(d.CustomValues, "/"))
```

produced clashes that the map -> json string of the same object avoids?

Reply:
Ah, saw your sample after I sent my question.

I'd personally lean towards this type of clash being exceedingly unlikely  - it can only happen between config-name, scope, and the custom values (and not for the domain and actionType custom values we have so far, as only domain is freely definable and can't start with a /).
And we could still choose another even less likely separator (e.g. `{%$`). 

But as this is only in the scope of delete file generation which runs on its own and probably not often in a scale that makes the memory and runtime overhead noticeable, I can live with it in it's safe but less efficient form.